test/test_pooling.py

```diff
@@ -379,6 +379,32 @@ def test_no_wait_queue_multiple(self):
         for socket_info in socks:
             socket_info.close_socket(None)
 
+    def test_maxConnecting(self):
+        client = rs_or_single_client()
+        self.addCleanup(client.close)
+        pool = get_pool(client)
+        docs = []
+
+        # Run 50 short running operations
+        def find_one():
+            docs.append(client.test.test.find_one({'$where': delay(0.001)}))
+        threads = [threading.Thread(target=find_one) for _ in range(50)]
+        for thread in threads:
+            thread.start()
+        for thread in threads:
+            thread.join(10)
+
+        self.assertEqual(len(docs), 50)
+        self.assertLessEqual(len(pool.sockets), 50)
```

Comment:
This test would be more useful if it asserted that we created way less than 50 connections, like this:
```python
        self.assertLessEqual(len(pool.sockets), 20)
```
But it would make the test more flaky, especially with noauth/nossl where connection creation is much faster. Hmm perhaps I can try changing this to:
```python
if ssl and auth enabled:
        self.assertLessEqual(len(pool.sockets), 20)
else:
        self.assertLessEqual(len(pool.sockets), 50)
```

Reply:
Done. Waiting on EVG test results to make sure this change doesn't make the test flakey.

Reply:
Nice test!